sdk/go/common/diag/colors/diag.go

```diff
@@ -48,10 +46,11 @@ func (c Colorization) ColorizeWithMaxWidth(v string, maxWidth int) string {
 		return colorizeText(v, Always, maxWidth)
 	case Never:
 		return colorizeText(v, Never, maxWidth)
-	default:
-		contract.Failf("Unexpected colorization value: %v", c)
-		return ""
+	case Auto:
```

Comment:
Does anyone know if `Auto` is actually used?  I would have expected problems here earlier if it was.

Reply:
No, there's some magic in `GetGlobalColorization` such that while Auto is the default value it's never actually used. I think I can tidy this up with an optional ptr rather than the Auto case.

Reply:
https://github.com/pulumi/pulumi/pull/15030